### PR TITLE
Remove exception from emailengine

### PR DESCRIPTION
**Pull request by smirnovaka**
remove runtime exception if builder is empty. Generate alarm.

### CODE REVIEW

File: src/foam/nanos/notification/email/EmailTemplateEngine.js [hunk 1]
```diff
@@ -67,7 +69,12 @@ foam.CLASS({
             if ( value == null ) {
               value = "";
               foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) x.get("logger");
-              logger.error("No value provided for variable " + val);
+              logger.warning("No value provided for variable " + v.toString());
```
Comment (kgrgreer) [inline src/foam/nanos/notification/email/EmailTemplateEngine.js]: The .toString() is unnecessary. Java will already do that.

File: src/foam/nanos/notification/email/EmailTemplateEngine.js [hunk 2]
```diff
@@ -67,7 +69,12 @@ foam.CLASS({
...
+              Alarm alarm = new Alarm();
+              alarm.setName("Email template config");
+              alarm.setReason(AlarmReason.CONFIGURATION);
+              alarm.setNote("No value provided for variable " + v.toString());
```
Comment (kgrgreer) [inline src/foam/nanos/notification/email/EmailTemplateEngine.js]: same

File: src/foam/nanos/notification/email/EmailTemplateEngine.js [hunk 3]
```diff
@@ -274,7 +281,12 @@ foam.CLASS({
             EmailTemplate extendedEmailTemplate = ((EmailTemplate) ((DAO) x.get("emailTemplateDAO")).find(EQ(EmailTemplate.NAME,templateName.toString())));
             if ( extendedEmailTemplate == null ) {
               foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) x.get("logger");
-              logger.error("Extended template not found " + val);
+              logger.warning("Extended template not found " + templateName.toString());
+              Alarm alarm = new Alarm();
+              alarm.setName("Email template config");
+              alarm.setReason(AlarmReason.CONFIGURATION);
+              alarm.setNote("Extended template not found " + templateName.toString());
```
Comment (kgrgreer) [inline src/foam/nanos/notification/email/EmailTemplateEngine.js]: same

Comment (smirnovaka) [reply src/foam/nanos/notification/email/EmailTemplateEngine.js]: thank you, updated.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**